### ISSUE TITLE
High Output Difference between ONNX model with different optimizer settings

### ISSUE THREAD

**Issue by okorokovnikita** [microsoft/onnxruntime#18959]
### Describe the issue

I have an ONNX model. When I compare the network outputs with different settings of optimizer, I observe a significant difference in the output tensors:

1. The heatmap below shows an output tensor with `ort.GraphOptimizationLevel.ORT_ENABLE_ALL`

![ENABLE_ALL](https://github.com/microsoft/onnxruntime/assets/57351674/f8431183-a909-4382-b15f-96691d3aff30)

2. The heatmap below shows an output tensor with `ort.GraphOptimizationLevel.ORT_ENABLE_EXTENDED`

![ENABLE_EXTENDED](https://github.com/microsoft/onnxruntime/assets/57351674/6c1671b5-cb82-485a-8ef2-59529f79c13c)

3. The heatmap below shows an output tensor with `ort.GraphOptimizationLevel.ORT_DISABLE_ALL`. Here we can see that heatmap of output tensor is completely different from the previous two heatmaps. And the range of tensor values is also different.

![DISABLE_ALL](https://github.com/microsoft/onnxruntime/assets/57351674/2014e2e2-e8bb-46cf-9bac-74e6dd3ad1b4)

Max absolute diff between output logits of model with `ort.GraphOptimizationLevel.ORT_ENABLE_ALL` and model with `ort.GraphOptimizationLevel.ORT_DISABLE_ALL`: 61.765

Mean absolute diff between output logits of model with `ort.GraphOptimizationLevel.ORT_ENABLE_ALL` and model with `ort.GraphOptimizationLevel.ORT_DISABLE_ALL`: 6.634

4. We tried to disable different optimizers and found out that disabling `LayerNormFusion` and `SimplifiedLayerNormFusion `optimizers at the same time is leading to broken tensor outputs.

![DISABLE_LAYERNORMFUSION_AND_SIMPLIFIEDLAYERNORMFUSION](https://github.com/microsoft/onnxruntime/assets/57351674/e43898d0-271b-4935-b469-f05fb799b23f)

The heatmap below shows a difference between output tensors with `ort.GraphOptimizationLevel.ORT_ENABLE_ALL` and disabled `LayerNormFusion` and `SimplifiedLayerNormFusion`.

![ENABLE_ALL_VS_DISABLE_LAYERNORMFUSION_AND_SIMPLIFIEDLAYERNORMFUSION](https://github.com/microsoft/onnxruntime/assets/57351674/26a9a00a-a93d-477c-94df-d836a2215af6)

Max absolute diff between output logits of model with `ort.GraphOptimizationLevel.ORT_ENABLE_ALL` and model with disabled `LayerNormFusion` and `SimplifiedLayerNormFusion`: 61.765

Mean absolute diff between output logits of model with `ort.GraphOptimizationLevel.ORT_ENABLE_ALL` and model with disabled `LayerNormFusion` and `SimplifiedLayerNormFusion`: 6.634

5. However disabling `LayerNormFusion` and `SimplifiedLayerNormFusion` seperatly does not break tensor outputs.

![DISABLE_LAYERNORMFUSION](https://github.com/microsoft/onnxruntime/assets/57351674/16c77285-8093-43fe-bdb4-14557bd1c7a3)

![DISABLE_SIMPLIFIEDLAYERNORMFUSION](https://github.com/microsoft/onnxruntime/assets/57351674/dadfb818-ef94-47fe-8efc-f739652efd5b)

I think this is a bug when two optimizers `LayerNormFusion` and `SimplifiedLayerNormFusion` are disabled at the same time.

### To reproduce

To reproduce a bug I put a model and input tensors [here](https://github.com/okorokovnikita/ort_optimizer_bug/tree/main). To reproduce results you can use jupyter notebook `issue_with_optimizers.ipynb` or run the script `python compare_enabled_all_vs_disabled_optimizers_outputs.py.`

### Urgency

Not urgent

### Platform

Linux

### OS Version

Ubuntu 20.04

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

onnxruntime==1.9.0

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA Version: 12.0

**Comment by liruichao-eon** [comment]
Similar issue here. Can anyone help to take a look? Thx! [OP (Conv) inference results mismatch with PyTorch #18946](https://github.com/microsoft/onnxruntime/issues/18946)
